### PR TITLE
Add integration tests for createBoard

### PR DESCRIPTION
**Pull request by chenilim**
#### Summary
Adding integration tests for createBoard, which currently fail. I'll file bugs to cover the failures, which will require fixes to the API.

### CODE REVIEW

File: server/integrationtests/board_test.go [hunk 1]
```diff
@@ -150,6 +150,28 @@ func TestCreateBoard(t *testing.T) {
...
+		})
+
+		t.Run("another user should be able to access the public board and its blocks", func(t *testing.T) {
+			rbBoard, resp := th.Client2.GetBoard(board.ID, "")
+			th.CheckOK(resp)
+			require.NotNil(t, rbBoard)
+			require.Equal(t, board, rbBoard)
+
+			rBlocks, resp := th.Client2.GetBlocksForBoard(board.ID)
+			th.CheckOK(resp)
```
Comment (chenilim) [inline server/integrationtests/board_test.go:172]: This test is currently failing. The GetBlocks API is not checking for team access on public  boards.

File: server/integrationtests/board_test.go [hunk 2]
```diff
@@ -241,6 +284,120 @@ func TestCreateBoard(t *testing.T) {
...
+			require.Equal(t, board, rbBoard)
+
+			rBlocks, resp := th.Client.GetBlocksForBoard(board.ID)
+			th.CheckOK(resp)
+			require.NotNil(t, rBlocks)
+		})
+
+		t.Run("unauthorized user should not be able to access the private board template or its blocks", func(t *testing.T) {
+			rbBoard, resp := th.Client2.GetBoard(board.ID, "")
+			th.CheckForbidden(resp)
```
Comment (chenilim) [inline server/integrationtests/board_test.go:391]: This test is currently failing because GetBlocks is allowing all users access to board templates.

**Comment by chenilim** [comment]
My intention of filing this PR is to validate the fixes for the two linked issues above. Let's fix those, then merge this after it passes.

**Comment by chenilim** [comment]
I've updated these tests with the latest API permissions behavior. Passing CI now.

**Review by jespino — APPROVED** [review]
I love this kind of additions :)

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Pull request merged**